lib/tx.py

```diff
@@ -78,6 +78,7 @@ class Deserializer(object):
     def __init__(self, binary, start=0):
         assert isinstance(binary, bytes)
         self.binary = binary
+        self.binary_length = len(binary)
```

Comment:
I'm not a fan of this one and from your timings the effect is marginal compared to the others.  Can we drop the len() optimisation please.